corgidrp/combine.py

```diff
@@ -0,0 +1,91 @@
+    n_samples = np.ones(data_subset.shape)
+    n_samples[bad] = 0
+    n_samples = np.sum(n_samples, axis=0)
+    if collapse.lower() == "mean":
+        data_collapse = np.nanmean(data_subset, axis=0) * tot_frames
+        err_collapse = np.sqrt(np.nanmean(err_subset**2, axis=0)) * tot_frames/np.sqrt(n_samples) # not sure if this is correct, but good enough for now
+    elif collapse.lower() == "median":
+        data_collapse = np.nanmedian(data_subset, axis=0) * tot_frames
+        err_collapse = np.sqrt(np.nanmean(err_subset**2, axis=0)) * tot_frames/np.sqrt(n_samples) * np.sqrt(np.pi/2) # inflate median error
+    # dq collpase: keep all flags on
+    dq_collapse = np.bitwise_or.reduce(dq_subset, axis=0)
+    # except those pixels that have been replaced 
+    dq_collapse[np.where((dq_collapse > 0) & (~np.isnan(data_collapse)))] = 0
+
+    return data_collapse, err_collapse, dq_collapse
+
+
+
+def combine_subexposures(input_dataset, num_frames_per_group=None, collapse="mean"):
+    """
+    Combines a sequence of exposures assuming a constant nubmer of frames per group
+    The length of the dataset must be divisible by the number of frames per group
+    
+    Args:
+        input_dataset (corgidrp.data.Dataset): input data. 
+        num_frames_per_group (int): number of subexposures per group. If None, combines all images together
+        collapse (str): "mean" or "median". Regardless, the images are scaled by num_frames_per_group to ~conserve photons
+
+    Returns:
+        corgidrp.data.Dataset: dataset after combination of every "num_frames_per_group" frames together
+    """
+    if num_frames_per_group is None:
+        num_frames_per_group = len(input_dataset)
+
+    if len(input_dataset) % num_frames_per_group != 0:
+        raise ValueError("Input dataset of lenght {0} cannot be grouped in sets of {1}".format(len(input_dataset, num_frames_per_group)))
+    
+    if collapse.lower() not in ["mean", "median"]:
+        raise ValueError("combine_subexposures can only collapse with mean or median")
+
+    num_groups = len(input_dataset) // num_frames_per_group
+    new_dataset = []
+    for i in range(num_groups):
+        data_subset = np.copy(input_dataset.all_data[num_frames_per_group*i:num_frames_per_group*(i+1)])
+        err_subset = np.copy(input_dataset.all_err[num_frames_per_group*i:num_frames_per_group*(i+1)])
+        dq_subset = input_dataset.all_dq[num_frames_per_group*i:num_frames_per_group*(i+1)]
+
+        data_collapse, err_collapse, dq_collapse = combine_images(data_subset, err_subset, dq_subset, collapse=collapse)
+
+        # grab the headers from the first frame in this sub sequence
+        pri_hdr = input_dataset[num_frames_per_group*i].pri_hdr.copy()
+        ext_hdr = input_dataset[num_frames_per_group*i].ext_hdr.copy()
+        err_hdr = input_dataset[num_frames_per_group*i].err_hdr.copy()
+        dq_hdr = input_dataset[num_frames_per_group*i].err_hdr.copy()
```

Comment:
Should we also be grabbing the hdu_list if it exists?

Reply:
We were doing it right below, but I've moved it up here to be more explicit.

Reply:
Whoops, I missed it, my bad!